site/docs/recipes/environment-variables.md

```diff
@@ -0,0 +1,279 @@
+export DATABASE_URL=postgres://user:pass@host:5432/db
+export API_KEY=your-secret-key
+export NODE_ENV=production
+
+preevy up
+```
+
+### In your CI environment
+
+Environment variables set in your CI environment are automatically passed through:
+
+#### GitHub Actions
+
+```yaml
+name: Deploy Preevy environment
+
+on:
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      
+      - uses: livecycle/preevy-up-action@v2.4.0
+        env:
+          DATABASE_URL: ${{ secrets.DATABASE_URL }}
+          API_KEY: ${{ secrets.API_KEY }}
+          NODE_ENV: production
+```
+
+#### Other CI systems
+
+Most CI systems allow you to set environment variables that will be available to Preevy:
+
+```bash
+# Environment variables are automatically available
+preevy up
+```
+
+## Environment-specific configurations
+
+You can create different configurations for different environments:
+
+### Using environment-specific .env files
+
+```yaml
+services:
+  web:
+    env_file:
+      - .env.${ENVIRONMENT:-development}
+```
+
+Then set the `ENVIRONMENT` variable:
+
+```bash
+# For staging
+export ENVIRONMENT=staging
+preevy up
+
+# For production
+export ENVIRONMENT=production
+preevy up
+```
+
+### Using conditional values in Compose files
+
+```yaml
+services:
+  web:
+    environment:
+      - NODE_ENV=${NODE_ENV:-development}
+      - LOG_LEVEL=${LOG_LEVEL:-info}
+      - DATABASE_URL=${DATABASE_URL:-sqlite://localhost/dev.db}
+  
+  database:
+    image: ${DB_IMAGE:-postgres:13}
+    environment:
+      - POSTGRES_DB=${POSTGRES_DB:-myapp_dev}
+      - POSTGRES_USER=${POSTGRES_USER:-dev_user}
+```
+
+## Best practices
+
+### 1. Use default values for development
+
+Always provide sensible defaults for local development:
+
+```yaml
+environment:
+  - API_URL=${API_URL:-http://localhost:3000}
+  - DATABASE_URL=${DATABASE_URL:-sqlite://localhost/dev.db}
+```
+
+### 2. Keep secrets out of compose files
+
+Never hardcode secrets in your compose files. Always use environment variables:
+
+```yaml
+# ❌ Don't do this
+environment:
+  - API_SECRET=hardcoded-secret
+
+# ✅ Do this
+environment:
+  - API_SECRET=${API_SECRET}
+```
+
+### 3. Document required environment variables
+
+Create a `.env.example` file to document required variables:
+
+```bash
+# .env.example
+DATABASE_URL=postgres://user:password@host:5432/database
+API_SECRET=your-secret-key-here
+NODE_ENV=development
+DEBUG=false
+```
+
+### 4. Use environment-specific configurations
+
+For different environments, use different approaches:
+
+```yaml
+# Development: use .env file with defaults
+env_file:
+  - .env
+
+# Staging/Production: use host environment variables
+environment:
+  - DATABASE_URL=${DATABASE_URL}
+  - API_SECRET=${API_SECRET}
+```
+
+## Troubleshooting
+
+### Environment variables not being interpolated
+
+If environment variables aren't being substituted, check:
```

Comment:
No need to be strongly opinionated about the practices, there's docker compose documentation for that, try to emphasis simple common simple cases, preevy is used for preview environments not production environments.